src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -794,6 +818,18 @@ public static string Escape(string argument)
             return escaped;
         }
 
+        public static string EscapeQuoted(string argument)
```

Comment:
We should be extra careful with `Regex` due to performance and testability issues. 
First, is there a OOTB system API available for that?
If `Regex` cannot be avoided consider using compiled instance.
Then, write tests covering all possible input combinations.